### PR TITLE
Add compute_path.py script to wrap used defines with ifndef macros

### PR DESCRIPTION
**Pull request by mmuesly**
Automated computation of header patches for CBMC proofs

Description
-----------
Checking configurations of define values requires to overwrite some of the configurations specified in the used config header files. Therefore, this script scans through all macro values used in CBMC proofs and computes a patch wrapping any define statement affected into an ifndef statement.

Checklist:
----------

- [x ] I have tested my changes. No regression in existing tests.
- [ x] My code is Linted.

### CODE REVIEW

File: tools/cbmc/patches/patch.py [hunk 1]
```diff
@@ -3,24 +3,18 @@
...
-
-tmpdir = 'c:\\temp\\' if sys.platform == 'win32' else '/tmp/'
-for patchfile in patchfiles:
-    tmpfile = tmpdir + patchfile
-    print("Patch file: {}".format(patchfile))
-    with open(patchfile) as inp, open(tmpfile, "w", newline="\n") as outp:
-        outp.write(inp.read())
-    cwd = os.getcwd()
-    result = subprocess.run(["git", "apply", tmpfile], cwd='../../..')
+for tmpfile in glob("*.patch"):
```
Comment (markrtuttle) [inline tools/cbmc/patches/patch.py]: Check that this code works under both bash and visual studio command prompt.

Comment (mmuesly) [reply tools/cbmc/patches/patch.py]: I have run this code successfully in a 64-bit visual studio command prompt.

File: tools/cbmc/README.md [hunk 2]
```diff
@@ -55,6 +55,11 @@ Installing CBMC
 Setting up the proofs
 ---------------------
 
+Change into the `patches` directory. Run
```
Comment (mmuesly) [inline tools/cbmc/README.md]: @karkhaz

These lines are added to address your comment. As you say this is not sufficient, could you describe in more detail, what kind of user documentation enhancement you envision?

Comment (karkhaz) [reply tools/cbmc/README.md]: Thanks, I completely missed this for some reason!

**Comment by mmuesly** [comment]
closed in favor of  #653

**Pull request closed without merging**